src/js/components/Main/Home/Home.container.jsx

```diff
@@ -13,28 +13,31 @@ function HomeContainer(props) {
     props.getConcerts(getConcerts)
   });
 
-  function getConcerts(concerts) {
-    setConcerts(concerts)
-  }
+  const getConcerts = useCallback(
+      (concerts)=>{setConcerts(concerts)}
+      )
+
+  const similarConcerts = useCallback(
+      function (val) {
+        if (val.length > 1) {
+          return val ?
+              concerts.filter((i) => i.band.toUpperCase().includes(val.toUpperCase())) :
+              []
+        } else {
+          return val ?
+              concerts.filter((i) => i.band.toUpperCase().startsWith(val.toUpperCase())) :
+              []
+        }
+      }
+  );
+
 
-  function similarConcerts(val) {
-    if (val.length > 1) {
-      return val ?
-          concerts.filter((i) => i.band.toUpperCase().includes(val.toUpperCase())) :
-          []
-    } else {
-      return val ?
-          concerts.filter((i) => i.band.toUpperCase().startsWith(val.toUpperCase())) :
-          []
-    }
-  }
 
   function onInputChange(event) {
     setInputText(event.target.value);
     setHelpList(similarConcerts(event.target.value));
   }
```

Comment:
How to rewrite this function via "useContext"?

Reply:
Do you mean `useCallback`?